psi4/driver/driver_cbs.py

```diff
@@ -1518,43 +2145,62 @@ def cbs(func, label, **kwargs):
     for mc in JOBS:
         instructions += """   %12s / %-24s for  %s%s\n""" % \
             (mc['f_wfn'], mc['f_basis'] + " + options"*bool(mc['f_options']),
-             VARH[mc['f_wfn']][mc['f_wfn']], addlremark[ptype])
+             VARH[mc['f_wfn']][mc['f_component']], addlremark[ptype])
 
     #     Remove duplicate modelchem portion listings
```

Comment:
What is a modelchem portion, or a modelchem portion listing?

Reply:
Haha yes, this is one of those CBS idiosyncrasies.

cbs() builds several "job" lists: 
- The `NEED` list contains one element for each non-delta stage and two for each delta stage, and contains the information required to run the calculations (basis, wfn, options).
- The `GRAND_NEED` list contains the same number of elements, including all of the information in `NEED`, and also additional information about how the energy/gradient obtained using the information in `NEED` has to be transformed to then be summed in the CBS recipe, and things required to print it in tables (contracted basis string, the "stage" tag...)
- The `MODELCHEM` list is a subset of `GRAND_NEED` including only elements where the components of energy (in VARH) match. 
- `JOBS` starts as a full-blown copy of `MODELCHEM`, but is reduced to a "minimal set" of elements that is equivalent to `MODELCHEM` - for example, we avoid calculating the same MP2 energy twice, even if it's used on it's own in `corl` and as a lower energy in `delta`. `MODELCHEM` retains both elements, as it knows how they need to be summed up and extrapolated, `JOBS` only retains one of them (ideally)
- `JOBS_EXT` is then created, looking up which energy/gradient components can be populated using the calculations in `JOBS`
- calculations in `JOBS` are executed and the returned quantities are stored into `JOBS_EXT`
- the obtained quantitites then have to be back-filled into `GRAND_NEED`. This is somehow done in the loop tagged "Dont ask", which uses `JOBS_EXT` to populate `GRAND_NEED`

So to actually answer your question, a "listing" would be whatever is in the `NEED` element - a specification on how to obtain an energy (or gradient). `MODELCHEM` contains all "listings" explicitly, as derived from the CBS recipe, but some "listings" may be subsumed in others (HF energy in MP2, both of those in CCSD). To determine whether two "listings" match, the ``component`` (i.e. Psi4 variable as tagged in `VARH`), ``basis`` and ``options`` have to match - but the ``wfn`` doesn't have to (CCSD != HF).

Reply:
Very valuable. Unless @loriab is going to change this in DDD, can we include this in the code? Trying to keep the various job lists straight without this is dizzying.